src/components/editor/editor-component.ts

```diff
@@ -102,21 +108,38 @@ export const actions = {
     // TODO: This making an assumption that mirrorMarkEditor has been initialized. This
     // may not hold.
     _state.textEditor.mirrorMarkEditor.setValue(data)
+    _actions.setOriginalData(data)
+  },
+
+  setOriginalData: (data: any) => (_state: any, _actions: any) => {
+    const codeMirrorEditor = _state.textEditor.mirrorMarkEditor.cm
+    codeMirrorEditor.on('keyup', () => {
+      _actions.updateSaveIcon(codeMirrorEditor.getValue() !== data)
+    })
+    return{
+      originalData: data,
+    }
   },
 
   saveTextNote: () => async (_state: any, _actions: any) => {
     const data = _state.textEditor.mirrorMarkEditor.getValue()
     await writeNote(_state.node.id, NoteDataType.TEXT, data)
   },
 
-  setNode: (node: IGraphNodeData) => (_: any, _actions: any) => {
+  setNode: (node: IGraphNodeData) => (_state: any, _actions: any) => {
```

Comment:
_state is _ because the _ implies is not being used.

Reply:
Fixed.